include/deal.II/lac/precondition.h

```diff
@@ -432,8 +516,52 @@ class PreconditionRelaxation : public Subscriptor
      */
     unsigned int n_iterations;
 
+    /**
+     * This sets the range between the largest eigenvalue in the matrix and
+     * the smallest eigenvalue to be treated. If the parameter is set to a
+     * number less than 1, an estimate for the largest and for the smallest
+     * eigenvalue will be calculated internally. For a smoothing range larger
+     * than one, the preconditioner will act in the interval
+     * $[\lambda_\mathrm{max}/ \tt{smoothing\_range}, \lambda_\mathrm{max}]$,
+     * where $\lambda_\mathrm{max}$ is an estimate of the maximum eigenvalue
+     * of the matrix. A choice of <tt>smoothing_range</tt> between 5 and 20 is
+     * useful in case the preconditioner is used as a smoother in multigrid.
+     */
+    double smoothing_range;
 
-    /*
+    /**
+     * Maximum number of CG iterations performed for finding the maximum
+     * eigenvalue. If set to zero, no computations are performed. Instead, the
+     * user must supply a largest eigenvalue via the variable
+     * PreconditionRelaxation::AdditionalData::max_eigenvalue.
+     */
+    unsigned int eig_cg_n_iterations;
+
+    /**
+     * Tolerance for CG iterations performed for finding the maximum
+     * eigenvalue.
+     */
+    double eig_cg_residual;
+
+    /**
+     * Maximum eigenvalue to work with. Only in effect if @p
+     * eig_cg_n_iterations is set to zero, otherwise this parameter is
+     * ignored.
+     */
+    double max_eigenvalue;
+
+    /**
+     * Constraints to be used for the operator given. This variable is used to
+     * zero out the correct entries when creating an initial guess.
+     */
+    AffineConstraints<double> constraints;
+
+    /**
+     * Stores the preconditioner object that the Chebyshev is wrapped around.
+     */
+    EigenvalueAlgorithm eigenvalue_algorithm;
+
+    /**
      * Preconditioner.
      */
     std::shared_ptr<PreconditionerType> preconditioner;
```

Comment:
Given that the comments given here are the same as in the Chebyshev class (this looks like copy-paste), would it make sense to introduce a base class and let `PreconditionChebyshev::AdditionalData` inhert from that class? I fear that having the same members and same documentation two times will lead to divergence with hard-to-understand concepts in the future. Inheritance should also be completely backwards compatible.

Reply:
I have introduced a base class. `PreconditionRelaxation::AdditionalData` now adds `relaxation` and `n_iterations` and `PreconditionChebyshev::AdditionalData` `degree` and `polynomial_type`. What do you think?